### PR TITLE
[SEMI-MODULAR] Stasis/Surgery Tweaks + New Stasis Chem

### PR DESCRIPTION
**Pull request by Higgin**
## About The Pull Request

Stasis was changed to always apply numbing in a recent fix. While not what I intended with my earlier changes to make anesthetic more important, this was good QOL because of how stasis works with processing chems - the usual sources of numbing. My earlier changes made stasis preserve numbing effects. People had to have the numbing effect on them prior to going into stasis in order to benefit from it - because many people tried to apply numbing agents on patients while they were in stasis, they found their efforts did nothing; it was ultimately frustrating and a pixel-bitchy guess as to whether or not you'd managed to numb somebody. Having stasis always numb people fixed that (albeit with a few problems - it didn't apply a new, separate source of numbing if you already had one from a chem, meaning you actively got punished for trying to numb people prior to surgery.)

What this did, however, was largely get around the points of my earlier PR: vulnerability. Having stasis give numbing and having numbing give the same speed benefits as unconsciousness made it doubly easy to slap people on a surgery table and work on them without a second thought. It made it doubly easy to expect that this is how you'd be treated coming to med if you weren't already dead, and it made it much harder to say there was any reason for somebody to be unconscious for surgery. 

You put somebody on a surgery table and they 1) no longer decayed, 2) no longer processed chems, 3) no longer breathed, 4) could be worked on immediately at full speed with advanced surgeries.

~~That surgery table, by the way, could be made with one (1) sheet of iron and one (1) sheet of silver. Anywhere. In about ten seconds.~~ no cost changes, axed these

This had a lot of room for improvement going forward, in part by taking a step back - _here are the big changes._

- this PR introduces doctor-facing differences in surgery speed between simple numbing and anesthetic. **Unanesthetized surgery is slower than it used to be. Unconsciousness is better.** 

- **stasis effects still numb** (UI fix was put in to reflect this - it was eating the status effect icons even after my earlier fix,) but **operating tables no longer give stasis.** They also have a slightly more sane cost to build (10 sheets up from 1.)

- **stasis beds are now once again researchable and buildable. They can be used for all the basic surgeries.** They do not come default on maps except Icebox and Kilo; other maps use pods.

- Surgical drapes now give you a warning if your patient is going to feel it, and the penalty on mood for working on people in softcrit (x is barely conscious...) is the lighter penalty. 

- Adds ataraxydone, a stasis-inducing chem. The stasis effect only works on the prone or incapacitated. This chem applies a significant slowdown which ramps up over time, eventually causing drowsiness/unconsciousness, and kicks in faster while prone/incapacitated. It allows a degree of movement/activity under its own stasis effect (hotkeys are blocked as are most complex interactions) as part of the tradeoff to being able to ignore other chemical/environmental/ticking effects. Its impure reagent causes cloneloss as does its OD. It's also a pain in the ass to make, but some comes in roundstart vendors.

## How This Contributes To The Skyrat Roleplay Experience

#10088 has a lot of the same justification as this PR - vulnerability is good, and it benefits from mechanical tradeoffs that people can point to in-game. This PR deepens those and hopefully does a lot for medical QOL along the way.

The way this is being done (removing stasis from op tables and deepening the reasons to use numbing agents) has a few other benefits to it.
+It gets chem players more involved with the production of numbing agents and preservatives like formaldehyde for surgeries where decay threatens (or where the number of bodies is overwhelming the number of stasis units.)
+It gets research more involved in making a potentially very useful upgrade to medical. Like medical often relies on science to get a Self-Actualization Device made to fix people who've lost their whole bodies, upgrading stasis pods to beds (they use the same components; it is a straight upgrade) may effectively double the number of people able to do surgery at one time on most maps, and make it possible to do regular wound-tending in triage as sometimes comes up on higher-intensity rounds... but **not without interdepartmental cooperation.**
+Since advanced surgeries can't be done without the operating computers, it gives the dedicated surgeries more of a niche than they currently have and one set apart from triage. This means that for those advanced surgeries (and the better wound-tending surgeries,) free numbing from the table is not a given, and it's not as good as the nearby gas. You'll have to set it up in advance if you want it or insist upon it. 
+For newer medical players in particular: making operating tables and operating rooms not come with stasis increases the likelihood that patients will be seen first in triage/the treatment centers where the actual stasis setups are by default. I suspect this will increase the likelihood of newer players seeing others, being seen, and getting help if they're struggling to figure out systems that they otherwise might fiddle with for half an hour alone in an operating room. 
+For all people playing medical, and anyone trying to do surgery: having drapes let you know whether or not your patient is under the effect of an anesthetic/unconscious should both offer a good tip towards what you should be doing and much-needed feedback if you were trying to do it in the first place. 

**ON THE STASIS CHEM:**
This might merit a TM. Stasis is an incredibly powerful effect. A few of the things you get with it:
+the ability to ignore atmosphere (yall like chilling in space?)
+the ability to ignore breathing (vox and plasmemes rejoice)
+the ability to ignore a lot of ticking effects (what if we kissed next to the SM?)
+99% of chems stop affecting you
+wounds stop causing problems i.e. bleeding, bloodloss stops causing oxyloss, etc.

I've tried to balance for this through a few things:
+you don't get stasis unless you're prone or incapacitated (it's meant for stabilizing people/surgery)
+it slows you down to an eventual crawl if you have it in your system (i.e. there's a significant cost to pre-game with it)
+the mobile stasis effect is obnoxious to play with, i.e., you're blocked from using a lot of hotkeys, interacting with complex objects, etc. - it is not conducive to "crawl into space and chase the badman naked" gaming
+trying to stack up on it all at once has nasty OD effects
+the slowdown and eventual unconsciousness kick in twice as fast if you're benefitting from the stasis effect
+it causes lasting damage if used impure which can only be healed via cryo/rezadone
+it's not easy to make safely or pure

_If it's so busted, why do this in the first place?_
Seems like there might be some cool use-cases for it in spite of the limitations I've imposed, and it gets around the issue of surgery/stasis needs for vox/plasmamen.

On the code-side, I'm sure there are a lot of things that could have been done more parsimoniously. The story there is as follows: when I first tried making a chem that added stasis, I realized stasis immobilizes/handblocks by default. I didn't want that on a chem because it'd make it busted in combat. I also realized that most of the effects for stasis are checked across a wide range of life files, and adding specific checks for this chem would be a godawful pain and mess. Instead, I duplicated the stasis effect into a modular file, tweaked it to allow for movement, and made it so the helper that checks for stasis also considers it. I welcome any feedback to make this cleaner or neater. 

## Changelog

:cl:
balance: Completely unanesthetized surgery now takes longer than numbed surgery takes longer than surgery on the fully-unconscious. Achieving current 'max' surgery speeds without other buffs like sterilizer or alcohol now requires unconsciousness.
balance: Mood penalties from unanesthetized surgery are less severe on those in soft-crit.
balance: Removes stasis from operating tables.
add: Readds open stasis beds (the type you can use for surgery) as a machine board researched at cryostasis technology.
fix: The closable stasis pods should now be findable under the appropriate techfab category (medical machinery) along with their open counterparts.
fix: Stasis effects now accurately update the UI to reflect that they also provide numbing. 
qol: Drapes now give a warning if the patient is unanesthetized prior to starting surgery.
add: Ataraxydone, a stasis-inducing chem that only works on the prone/incapacitated, and which allows for limited activity under stasis. 
/:cl:

### CODE REVIEW

**Comment by Zergspower** [comment]
Can the surgery tables just please get an on/off for stasis as well, the stasis beds have it and the whole issue getting awake people to lay down is hilariously stupid to get the point across to them lol. This is assuming this gets changes from the coders if they dont like it - I do see it was removed entirely which im fine with (though plasmamen will be a pain to operate on)

File: code/game/objects/structures/tables_racks.dm
Comment (tf-4) [inline code/game/objects/structures/tables_racks.dm]: What's the edit here?

Comment (Higgin) [reply code/game/objects/structures/tables_racks.dm]: It was previously where thawing/chilling were set for stasis, but on a relook and retest it's redundant with the loc check. Removing it shortly.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
I haven't reviewed the code just yet, but I still wanted to leave a review before I forget.

I agree with pretty much all the changes in this PR, except making the surgery table more expensive, but I can see from quickly glancing at the code that it's not the case anymore, so probably fix the PR body to remove the mention of that change.

However, there's one extra change I want from this PR. Currently, the only advantage that I thought made sense from surgery tables was that it allowed surgery on the plasmamen. While I'm aware that showers exist, let's be honest here, it doesn't make so much sense that we literally have to put them under running water to operate on them when we have literal stasis technology.
That's why I'd like to also see a solution to that problem also being presented in this PR, would it be via a new chemical, or a new device that could be installed on the surgery beds to allow for an active plasma field that you turn on or off depending on if it's a plasmaman or not (with visual effects, damn this would be pretty cool, but that might be a bit much for this PR).

If you need help brainstorming it, hit me up on Discord and we can discuss it in #development-discuss.

File: code/modules/surgery/surgery_step.dm
Comment (Iamgoofball) [inline code/modules/surgery/surgery_step.dm:97]: /= is a cursed operator

File: modular_skyrat/modules/stasisrework/code/medical_designs.dm
Comment (Iamgoofball) [inline modular_skyrat/modules/stasisrework/code/medical_designs.dm:11]: can you roll these back to being stasis beds while you're here?

**Comment by Iamgoofball** [comment]
@GoldenAlpharex If Stasis Beds are back then we can properly handle plasmamen and vox again.

**Comment by Zergspower** [comment]
> @GoldenAlpharex If Stasis Beds are back then we can properly handle plasmamen and vox again.

it poses an RP problem I've had with since playing on the server, we shouldn't be doing surgery on people in an open room that everyone can walk into. A. privacy and B. sterilization. The stasis beds on the maps that have them are in the wide-ass open, one problem there can be solved by maybe a policy tweak but the handicap to those two raaces I think needs that additional solution in place

**Comment by Iamgoofball** [comment]
>it poses an RP problem I've had with since playing on the server, we shouldn't be doing surgery on people in an open room that everyone can walk into.

Triage isn't a public access room. Medbay can and will remove you from Triage if you're not a patient. No policy changes are needed.

**Comment by YakumoChen** [comment]
:+1: on making knockout preferable for surgery

:-1: on surgery tables needing 10 silver. That's bad for several balance implications, one of which is that you can decon tables for 10 silver roundstart instead of 1.

Roundstart implications are far more important than lateround when 10 silver means literally nothing and it might as well be 1 silver for whatever purpose. Making the number higher only has negative effects.

File: modular_skyrat/modules/stasisrework/code/medical_designs.dm
Comment (Higgin) [reply modular_skyrat/modules/stasisrework/code/medical_designs.dm:11]: Thought I might leave them (the boards are also available now) in case folks wanted to use them for aesthetics or for a stopgap before upgrading on the maps that still have them. Would you rather they just be ripped out?

**Comment by Higgin** [comment]
Going to hold off on squashing commits until this gets a close pass.
@YakumoChen cost is no longer changed.
@GoldenAlpharex Going with the chem angle on your request.

File: modular_skyrat/modules/modular_reagents/code/debuffs.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_reagents/code/debuffs.dm:11]: document this.

File: modular_skyrat/modules/modular_reagents/code/debuffs.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_reagents/code/debuffs.dm:13]: documentation please

File: modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm]: Don't colors work with a `#` in front of them? I forget.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
My brain works slowly today, there's also some grammar/capitalization issues with the to_chats when it comes to the metabolization but I'm too lazy to scroll back down to make suggestions.

File: modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm
Comment (Higgin) [reply modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm]: Worked without, but I put one in for consistency. Still seems good.

File: modular_skyrat/modules/modular_reagents/code/debuffs.dm
Comment (Higgin) [reply modular_skyrat/modules/modular_reagents/code/debuffs.dm:11]: I added comments on the file - what, exactly, are we looking for as far as documenting this goes?

File: code/__DEFINES/status_effects.dm
Comment (GoldenAlpharex) [inline code/__DEFINES/status_effects.dm:46]: Why is this made here and not where that define is originally at? Unless GitHub doesn't show it to me properly on mobile, this seems like just a pure addition.

Which at that point it could be put in the ~skyrat folder of the __DEFINES folder, so it's modular.

File: code/datums/elements/surgery_initiator.dm
Comment (GoldenAlpharex) [inline code/datums/elements/surgery_initiator.dm:1]: Put this in a module in the modular Skyrat folder, we shouldn't be adding modular files to non-modular folders.

File: code/datums/elements/surgery_initiator.dm
Comment (GoldenAlpharex) [inline code/datums/elements/surgery_initiator.dm:24]: I'm not going to make the suggestion bigger because fuck GitHub on mobile, but use snake_case to separate your words.
```suggestion
	var/mob/living/living_target = target
```

File: code/datums/elements/surgery_initiator.dm
Comment (GoldenAlpharex) [inline code/datums/elements/surgery_initiator.dm:34]: That's not a really descriptive variable name. Do something like `for(var/surgery in living_target.surgeries)`

File: code/datums/elements/surgery_initiator.dm
Comment (GoldenAlpharex) [inline code/datums/elements/surgery_initiator.dm:43]: My head hurts just looking at this loop, surely there's a way to outsource some of these checks to external procs to make this simpler, or something. Because man that's a lot of checks in a row.

File: code/datums/status_effects/debuffs.dm
Comment (GoldenAlpharex) [inline code/datums/status_effects/debuffs.dm:247]: I know it's unrelated to this PR, but I still wonder why this has stopped looping, like, it's meant to loop every so often but it just doesn't.

File: code/modules/surgery/surgery_step.dm
Comment (GoldenAlpharex) [reply code/modules/surgery/surgery_step.dm:97]: I feel like this could just be something that does *= 2 instead of /= 0.5
It's be cleaner code, even if it's a separate define.

File: code/modules/surgery/surgery_step.dm
Comment (GoldenAlpharex) [inline code/modules/surgery/surgery_step.dm:96]: I think this is what you meant to do, otherwise it would be really cursed.
Also in general I feel like you should do one check for the was_sleeping or if they're at least unconscious, I don't know, this feels like a lot of code repetition.
```suggestion
	if(!(was_sleeping || HAS_TRAIT(target, TRAIT_NUMBED) || target.stat < UNCONSCIOUS))
```

File: code/modules/surgery/surgery_step.dm
Comment (GoldenAlpharex) [inline code/modules/surgery/surgery_step.dm:214]: Is SOFT_CRIT above UNCONSCIOUS?

File: modular_skyrat/modules/modular_reagents/code/debuffs.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_reagents/code/debuffs.dm:11]: Something like this is what we call documentation for variables. Every time you have a `var/something` in a class definition (so, not inside of a proc), you should be documenting it with a short sentence saying what it does.

Disclaimer: I don't know if this is what that variable means, but I wanted to give you an example.
```suggestion
	/// Time since the person last died.
	var/last_dead_time
```

File: modular_skyrat/modules/modular_reagents/code/debuffs.dm
Comment (GoldenAlpharex) [reply modular_skyrat/modules/modular_reagents/code/debuffs.dm:13]: Documentation for a proc should at the minimum say what the proc does, and usually should say what each parameter represents/is used for, and finally what the proc returns, if it returns anything. I'm not on my computer so I can't give you a great example here, but for an example of what's the kind of documentation I *really* like, look at `conveyor.dm` (or maybe it's called `recycling.dm`, I don't remember the name).

File: modular_skyrat/modules/modular_reagents/code/debuffs.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_reagents/code/debuffs.dm:16]: Here again, snake_case please. I'm not going to comment it everywhere because it takes forever to leave comments on mobile.

File: modular_skyrat/modules/modular_reagents/code/modifiers/reagent.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_reagents/code/modifiers/reagent.dm:1]: This isn't how we do multiline comments, look up the Autodoc Comment Block extension on VSC, I can link it to you on Discord if I can find it. This is really funky and hurts my soul a little.

File: modular_skyrat/modules/modular_reagents/code/modifiers/reagent.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_reagents/code/modifiers/reagent.dm:4]: Documentation should always be using either `///` or `/* */` (for multiline ones), otherwise they won't be recognized by the autodoc function of the DM Lang server extension, or the AUTODOC workflow on the repo.

File: modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm:41]: It will process in dead people? Will that do like formaldehyde, or will it actually be metabolized while dead?

File: modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm:49]: Those numbers should be defines, I don't know what they stand for.

File: modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm:53]:
```suggestion
		current_cycle++
```

File: modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm:57]: This should be a define too. And those kinds of defines should be undefined at the end of the file, using `#undef DEFINE_NAME`

File: modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm:63]: Those values in the switch case should be defines too.

File: modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm:87]:
```suggestion
				to_chat(owner, span_notice("... What's the point of fighting, anyway?"))
```

File: modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm:88]: "ataraxydone" here should be a define, like `TRAIT_SOURCE_ATARAXYDONE_OD` (or something like that, I forget what the standard way of writing those source defines is, I recommend checking that yourself).

File: modular_skyrat/modules/modular_reagents/code/reagents/reagent_containers/syringes.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_reagents/code/reagents/reagent_containers/syringes.dm:4]: Will it also contain the impure chem in it? If so, should probably inform people on the syringe that its cellular damage will need to be dealt with after use, or something like that.

File: modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_reagents/code/reagents/medicine.dm:103]: How much cloning damage does that actually do?

File: modular_skyrat/modules/modular_reagents/code/recipes/medicine_recipes.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_reagents/code/recipes/medicine_recipes.dm:23]: Have you tried making that chem yourself during testing? Just how complicated is it to make, and how much of a pain is that reaction to keep in the pure range?

File: modular_skyrat/modules/stasisrework/code/medical_designs.dm
Comment (GoldenAlpharex) [reply modular_skyrat/modules/stasisrework/code/medical_designs.dm:11]: I personally never really liked them, due to the controls being really awkward and counter-intuitive, and because of how easy it is to forget someone in there.

I wouldn't be against us gutting those sleepers away, tbh. I'll talk with the other maintainers to see what we think about it.

File: modular_skyrat/modules/stasisrework/code/stasissleeper.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/stasisrework/code/stasissleeper.dm:2]: I hate how this didn't even follow the snake_case convention.

File: code/datums/elements/surgery_initiator.dm
Comment (GoldenAlpharex) [reply code/datums/elements/surgery_initiator.dm:1]: Also I don't even think it was ticked 💀

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
There's a bit more work that's left to put in, in terms of documentation (hopefully my examples help), modularization and testing, and also a bit of code improvement and following coding standards too.

I think we're getting close from a stage where we can test-merge this.

**Comment by Gandalf2k15** [comment]
Feel free to reopen with requested changes.

**Comment by GoldenAlpharex** [comment]
(or just ping me on Discord once you've made the changes, and I'll re-open it for you)

**Pull request closed without merging**
